ingestion-beam/src/test/java/com/mozilla/telemetry/transforms/PubsubMessageToTableRowTest.java

```diff
@@ -235,4 +238,76 @@ public void testNestedRepeatedStructs() throws Exception {
     assertEquals(expectedAdditional, Json.asString(additionalProperties));
   }
 
+  @Test
+  public void testRawFormat() throws Exception {
+    PubsubMessage message = new PubsubMessage("test".getBytes(),
+        // Use example values for all attributes present in the spec:
+        // https://github.com/mozilla/gcp-ingestion/blob/master/docs/edge.md#edge-server-pubsub-message-schema
+        ImmutableMap.<String, String>builder()
+            .put("submission_timestamp", "2018-03-12T21:02:18.123456Z")
+            .put("uri",
+                "/submit/telemetry/6c49ec73-4350-45a0-9c8a-6c8f5aded0cf/main/Firefox/58.0.2"
+                    + "/release/20180206200532")
+            .put("protocol", "HTTP/1.1").put("method", "POST").put("args", "v=4")
+            .put("remote_addr", "172.31.32.5").put("content_length", "4722")
+            .put("date", "Mon, 12 Mar 2018 21:02:18 GMT").put("dnt", "1")
+            .put("host", "incoming.telemetry.mozilla.org").put("user_agent", "pingsender/1.0")
+            .put("x_forwarded_for", "10.98.132.74, 103.3.237.12").put("x_pingsender_version", "1.0")
+            .put("x_debug_id", "my_debug_session_1")
+            .put("x_pipeline_proxy", "2018-03-12T21:02:18.123456Z").build());
+    String expected = Json.asString(ImmutableMap.<String, String>builder()
+        .putAll(message.getAttributeMap()).put("payload", "dGVzdA==").build());
```

Comment:
The payload is showing up as base-64 encoded in the json. Do we understand where that encoding is happening? Are we sure that what we end up being inserted into BigQuery is the bytes rather than the encoded string? We should probably add a new case in BigQueryIntegrationTest to validate we get what we expect in a BYTES column.

Reply:
> Do we understand where that encoding is happening?

`Json.asString(tableRow)` will encoded a bytes field as base64, because bytes isn't a valid json type

> We should probably add a new case in BigQueryIntegrationTest to validate we get what we expect in a BYTES column.

good point